src/pc/tracker.rs

```diff
@@ -425,6 +434,21 @@ impl SubxtTracker {
 				.await?,
 		);
 
+		if let Some((relay_block_number, relay_block_hash)) = self.current_relay_block {
+			let maybe_relevant_finalized_block_number = self
+				.api
+				.storage()
+				.storage_read_prefixed(CollectorPrefixType::RelevantFinalizedBlockNumber, relay_block_hash)
+				.await;
+			self.current_finality_lag = match maybe_relevant_finalized_block_number {
+				Some(entry) => {
+					let relevant_finalized_block_number: u32 = entry.into_inner()?;
```

Comment:
We shouldn't return error here as we wouldn't process this block inherent at all. We can just log warn and continue without finality lag.

Reply:
Make sense, thank you